core/src/main/java/org/stellar/anchor/util/SepLanguageHelper.java

```diff
@@ -0,0 +1,105 @@
+package org.stellar.anchor.util;
+
+import static org.stellar.anchor.util.Log.errorEx;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import org.stellar.anchor.api.exception.SepValidationException;
+import org.stellar.anchor.config.AppConfig;
+
+public class SepLanguageHelper {
+  static Map<String, Iso3316Language> fallbackLangs = null;
+
+  public static String validateLanguage(AppConfig appConfig, String lang) {
+    if (fallbackLangs == null) {
+      fallbackLangs = prepareFallbackLanguages(appConfig);
+    }
+
+    if (lang != null) {
+      List<String> languages = appConfig.getLanguages();
+      if (languages != null && languages.size() > 0) {
+        if (languages.stream().noneMatch(l -> l.equalsIgnoreCase(lang))) {
+          return getFallbackLanguage(lang);
+        }
+      }
+      return lang;
+    } else {
+      return getFallbackLanguage(lang);
+    }
+  }
+
+  public static void reset() {
+    fallbackLangs = null;
+  }
+
+  static Map<String, Iso3316Language> prepareFallbackLanguages(AppConfig appConfig) {
+    Map<String, Iso3316Language> fallbacks = new HashMap<>();
+    appConfig
+        .getLanguages()
+        .forEach(
+            lang -> {
+              try {
+                Iso3316Language isoLang = Iso3316Language.of(lang);
+                fallbacks.putIfAbsent(isoLang.getLangKey(), isoLang);
```

Comment:
❓  Won't languages with the same prefix be overwritten? Like, if you had both `"zh-TW"` and `"zh-CN"`, wouldn't one of them be left out if you use `getLangKey`? Maybe you should use `toString()` instead.

Reply:
The lang key defines how we should find the alternative language. Here is the language finding strategy:

1. Find the exact match. In this case, `zh-TW` will only match `zh-TW`
2. Find the alternative match. In this case, `zh-TW` will find some language that starts with `zh`. And that alternative language is the first occurrence in the `languages` list. (this is why the code is written this way)
3. Fallback to DEFAULT_LANGUAGE_WITHOUT/WITH_LOCALE_DEFAULT